packages/react-router/modules/Redirect.js

```diff
@@ -49,6 +51,24 @@ class Redirect extends React.Component {
       this.perform()
   }
 
+  componentDidUpdate(prevProps) {
+    let prevTo = prevProps.to
+    let nextTo = this.props.to
+
+    prevTo = typeof prevTo === 'string' ? { pathname: prevTo } : prevTo
```

Comment:
@pshrmn Turns out `createLocation` was exactly what I needed, thanks!  
Added a test for this use case.